lib/ical/duration.js

```diff
@@ -273,8 +273,20 @@ class Duration {
       let str = "";
       if (this.isNegative) str += "-";
       str += "P";
-      if (this.weeks) str += this.weeks + "W";
-      if (this.days) str += this.days + "D";
+      console.log(this.weeks)
+      if(this.weeks){
+        if (this.days || this.hours || this.minutes || this.seconds){
+          str += (this.weeks*7 + this.days + "D");
+        }
+        else{
+          str += (this.weeks + "W");
+        }
+      }
+      else{
+        if(this.days){
+          str += (this.days + "D");
+        }
+      }
 
       if (this.hours || this.minutes || this.seconds) {
```

Comment:
Looking at the linked issue it sounds like dur-week is mutually exclusive with dur-time. If we want to produce valid values we should probably also take care of this.